server/src/main/java/org/opensearch/cluster/routing/OperationRouting.java

```diff
@@ -300,7 +338,18 @@ private ShardIterator shardRoutings(
         @Nullable ResponseCollectorService collectorService,
         @Nullable Map<String, Long> nodeCounts
     ) {
-        if (ignoreAwarenessAttributes()) {
+        WeightedRoutingMetadata weightedRoutingMetadata = null;
+        if (clusterService != null) {
+            weightedRoutingMetadata = clusterService.state().metadata().weightedRoutingMetadata();
+        }
+        if (weightedRoutingMetadata != null) {
```

Comment:
For clarification we would have a `DELETE` API to remove the metadata right?

Reply:
Yeah, we do have delete API. The changes are in another PR - https://github.com/opensearch-project/OpenSearch/pull/4400/commits/8ce52556a0718e0b464776715797830065c0ab8d